libcodechecker/server/api/report_server.py

```diff
@@ -761,17 +770,19 @@ def getDiffResultsHash(self, run_ids, report_hashes, diff_type):
                     return new_hashes
             elif diff_type == DiffType.RESOLVED:
                 results = session.query(Report.bug_id) \
-                    .filter(Report.run_id.in_(run_ids)) \
-                    .filter(Report.bug_id.notin_(report_hashes)) \
-                    .all()
+                    .filter(Report.bug_id.notin_(report_hashes))
+
+                if run_ids:
+                    results = results.filter(Report.run_id.in_(run_ids))
 
                 return [res[0] for res in results]
```

Comment:
shouldn't we query all the results here?
`[res[0] for res in results]` -> `[res[0] for res in results.all()]`